### DISCUSSION TITLE
tsx code highlight and linting in HTML

### DISCUSSION THREAD

**Discussion by ChenZhongPu** [LazyVim/LazyVim#2826] [Q&A]
How to highlight and lint tsx code in HTML?

![image](https://github.com/LazyVim/LazyVim/assets/2310958/1dd84eed-19fe-4eed-9c1a-3b937e5513b7)

To put it in another way, is it possible to let `tsserver` work for html?

**Comment by Demianeen** [comment]
It works for me without any additional setup
![CleanShot 2024-03-26 at 14 09 27@2x](https://github.com/LazyVim/LazyVim/assets/51330172/5f84d303-c695-43ae-950b-93f2e0239314)

Do you have treesiter jsx installed?

Try with:
```lua
  {
    'nvim-treesitter/nvim-treesitter',
    opts = function(_, opts)
      if type(opts.ensure_installed) == 'table' then
        vim.list_extend(
          opts.ensure_installed,
          { 'typescript', 'tsx', 'embedded_template' }
        )
      end
    end,
  },
```

**Reply by ChenZhongPu** [reply]
The `typescript` and `tsx` are installed by default using lazyvim [starter](https://github.com/LazyVim/starter). Is `embedded_template` necessary to highlight tsx code in HTML?

**Reply by Demianeen** [reply]
The tsx parser should do the trick. But can you try to install them manually? Because for me even with all extras and my plugins disabled it works just fine. I don't know otherwise, maybe it because I use nightly but it sounds unrelated

I mean use `:TSInstall tsx` and restart nvim

**Reply by ChenZhongPu** [reply]
`tsx` is installed already. But it still does not work as expected in HTML.

<img width="490" alt="image" src="https://github.com/LazyVim/LazyVim/assets/2310958/f464bd54-5d1f-4ed1-9a23-62b76eabc722">
